cmd/api/src/api/middleware/middleware.go

```diff
@@ -277,3 +278,22 @@ func FeatureFlagMiddleware(db database.Database, flagKey string) mux.MiddlewareF
 		})
 	}
 }
+
+func EnsureRequestBodyClosed() mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+			next.ServeHTTP(response, request)
+
+			switch b := request.Body.(type) {
```

Comment:
Recommending that we capture the reason this odd code is here in a comment. This is running into the nil != nil problem in Go, and the underlying cause is in stdlib interactions, so this seems like our only real option. Code-wise, the solution is solid we just want historical information available for the next poor engineer that needs to maintain this in the future.